src/expire.c

```diff
@@ -161,16 +174,112 @@ static inline int isExpiryTableValidForSamplingCb(hashtable *ht) {
     return C_OK;
 }
 
-void activeExpireCycle(int type) {
+static inline int activeExpireFieldsCheckTimeLimitReached(
+    unsigned int *iterations,
+    uint64_t start_us,
+    uint64_t limit_us,
+    uint64_t *now_us) {
+    if (((*iterations)++ & 0xf) == 0) {
+        *now_us = ustime();
+    }
+    return (*now_us - start_us >= limit_us);
+}
+
+
+void advanceDb(activeExpireFieldIterator *it) {
+    it->current_db++;
+    if (it->current_db >= server.dbnum) {
+        it->current_db = 0;
+        it->db_cursor = 0;
+    }
+    it->current_key = NULL;
+}
+
+static inline int effort(void) {
+    return server.active_expire_effort - 1;
+}
+
+void hashKeyDone(activeExpireFieldIterator *it) {
```

Comment:
hashKeyDone what? also why focus only on `hash`? in case we add more types (e.g sets ) we might have more types. suggest to name it more explicitly e.g activeExpireKeyDone